sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -865,23 +866,23 @@ qualifiedColTypeWithPositionList
     ;
 
 qualifiedColTypeWithPosition
-    : name=multipartIdentifier dataType (COMMENT comment=STRING)? colPosition?
+    : name=multipartIdentifier dataType (NOT NULL)? (COMMENT comment=STRING)? colPosition?
     ;
 
 colTypeList
     : colType (',' colType)*
     ;
 
 colType
-    : colName=errorCapturingIdentifier dataType (COMMENT STRING)?
+    : colName=errorCapturingIdentifier dataType (NOT NULL)? (COMMENT comment=STRING)?
```

Comment:
What behavior we will see if we set NOT NULL in v1 queries? Failure or silently ignoring?

Reply:
it depends on data source. You can see how we determine the schema in `CreateDataSourceTableCommand`.

BTW this is an existing problem: users can call `spark.catalog.createTable` with non-nullable schema.